sambacc/ctdb.py

```diff
@@ -490,17 +488,96 @@ def cluster_meta_to_nodes(cmeta: ClusterMeta, real_path: str) -> None:
         json_data = cmo.load()
         nodes = json_data.get("nodes", [])
         _logger.info("Found node metadata: %r", nodes)
-        pnn_max = max(n["pnn"] for n in nodes) + 1  # pnn is zero indexed
-        ctdb_nodes: list[str] = [""] * pnn_max
-        for entry in nodes:
-            pnn = entry["pnn"]
-            expected_line = _entry_to_node(ctdb_nodes, entry)
-            ctdb_nodes[pnn] = expected_line
+        ctdb_nodes = _cluster_meta_to_ctdb_nodes(nodes)
         _logger.info("Will write nodes: %s", ctdb_nodes)
-        with open(real_path, "w") as nffh:
-            write_nodes_file(nffh, ctdb_nodes)
-            nffh.flush()
-            os.fsync(nffh)
+        _save_nodes(real_path, ctdb_nodes)
+
+
+def _cluster_meta_to_ctdb_nodes(nodes: list[dict]) -> list[str]:
+    pnn_max = max(n["pnn"] for n in nodes) + 1  # pnn is zero indexed
+    ctdb_nodes: list[str] = [""] * pnn_max
+    for entry in nodes:
+        pnn = entry["pnn"]
+        expected_line = _entry_to_node(ctdb_nodes, entry)
+        ctdb_nodes[pnn] = expected_line
```

Comment:
Consider making it a one-liner:
`ctdb_nodes[pnn] = _entry_to_node(ctdb_modes,  entry)`

Reply:
ok, done in a separate commit to preserve the move-only nature of the original commit.